CONTRIBUTING.md

```diff
@@ -0,0 +1,223 @@
+
+For example, our fee calculator is typed as follows:
+```ts
+// In '.d.ts' file
+type DepositFeeCalculator = {
+  onChainDepositFee({ amount, ratio }: onChainDepositFeeArgs): Satoshis
+  lnDepositFee(): Satoshis
+}
+
+// In implementation file
+export const DepositFeeCalculator = (): DepositFeeCalculator => {
+  const onChainDepositFee = ({ amount, ratio }: onChainDepositFeeArgs) => {
+    return toSats(Math.round(amount * ratio))
+  }
+
+  return {
+    onChainDepositFee,
+    lnDepositFee: () => toSats(0),
+  }
+}
+```
+
+Note that the top-level function arguments (for `DepositFeeCalculator`) would still be typed at the point of implementation like with any other function, and it is only the method signatures that are included in declaration files.
+
+#### Interfaces
+We use the `interface` keyword to define objects with methods that are intended to be implemented outside of the domain. This most often happens with service implementations like our `ILightningService` interface that gets implemented as the `LndService` function.
+
+Everything else about how we go about typing and implementing things with the `interface` keyword is the same as with how we handle "objects with methods" in our domain as described above.
+
+#### "Enums" via object constants
+Typescript's `enum` keyword has drawbacks that don't fully meet our typing needs. To get around this, we use the trick where we define our intended enum as a standard object and then import it into our type declarations.
+
+For example:
+```ts
+// In implementation file
+export const AccountStatus = {
+  Locked: "locked",
+  Active: "active",
+} as const
+
+// In declaration file
+type AccountStatus =
+  typeof import("./index").AccountStatus[keyof typeof import("./index").AccountStatus]
+```
+## Architecture
+We use hexagonal architecture pattern (some link). 
+
+Code goes into one of four layers:
+- Domain layer (`./src/domain`)
+- Services layer (`./src/services`)
+- Application layer (`./src/app`)
+- Application Access layer (`./src/servers`)
+
+### Domain layer
+Defines the models and business logic with related interfaces.
+
+#### Responsibility
+- Define all data types
+- Implement operations on the data types that depends on conditionals or data transformations
+- Define interfaces of external services
+
+#### Dependencies
+- Internal: None
+- External: only utility libraries but must be as clean as possible
+
+
+### Services layer
+Implements all the adapters (specific implementations of the interfaces defined in domain layer) that use external services/resources
+
+#### Responsibility
+- Implement interfaces defined in domain layer
+
+#### Examples
+- Access to external resources (database, redis, bitcoind, lnd)
+- Consumption of external services/APIs (twilio, geetest, price, hedging)
+
+#### Dependencies
+- Internal: Domain Layer
+- External: all required dependencies
+
+
+### Application layer
```

Comment:
`Application access layer` versus `Application layer` is not clear to me the difference -> naming can probably be improved

Reply:
Sure, this whole section is probably incomplete though. I believe our intention here is to commit what we have for now and then come back to finishing off this document instead of leaving it as a pending PR.